cpp/example_code/dynamodb/query_items.cpp

```diff
@@ -1,130 +1,159 @@
+            request.SetExclusiveStartKey(exclusiveStartKey);
+            exclusiveStartKey.clear();
+        }
         // Perform Query operation
-        const Aws::DynamoDB::Model::QueryOutcome& result = dynamoClient.Query(req);
-        if (result.IsSuccess())
-        {
+        const Aws::DynamoDB::Model::QueryOutcome &outcome = dynamoClient.Query(request);
+        if (outcome.IsSuccess()) {
             // Reference the retrieved items
-            const Aws::Vector<Aws::Map<Aws::String, Aws::DynamoDB::Model::AttributeValue>>& items = result.GetResult().GetItems();
-            if(items.size() > 0) 
-            {
-                std::cout << "Number of items retrieved from Query: " << items.size() << std::endl;
+            const Aws::Vector<Aws::Map<Aws::String, Aws::DynamoDB::Model::AttributeValue>> &items = outcome.GetResult().GetItems();
+            if (!items.empty()) {
+                std::cout << "Number of items retrieved from Query: " << items.size()
+                          << std::endl;
                 //Iterate each item and print
-                for(const auto &item: items)
-                {
-                std::cout << "******************************************************" << std::endl;
-                // Output each retrieved field and its value
-                for (const auto& i : item)
-                    std::cout << i.first << ": " << i.second.GetS() << std::endl;
+                for (const auto &item: items) {
+                    std::cout
+                            << "******************************************************"
+                            << std::endl;
+                    // Output each retrieved field and its value
+                    for (const auto &i: item)
+                        std::cout << i.first << ": " << i.second.GetS() << std::endl;
                 }
             }
-
-            else
-            {
-                std::cout << "No item found in table: " << table << std::endl;
+            else {
+                std::cout << "No item found in table: " << tableName << std::endl;
             }
+
+            exclusiveStartKey = outcome.GetResult().GetLastEvaluatedKey();
         }
-        else
-        {
-            std::cout << "Failed to Query items: " << result.GetError().GetMessage();
+        else {
+            std::cerr << "Failed to Query items: " << outcome.GetError().GetMessage();
+            result = false;
+            break;
         }
-        // snippet-end:[dynamodb.cpp.query_items.code]
+    } while (!exclusiveStartKey.empty());
+
+    return result;
+}
+// snippet-end:[dynamodb.cpp.query_items.code]
+
+/*
+ *  main function
+ *
+ *  Usage: 'run_query_items <table_name> <partition_key> <partition_value> [projection_expression]'
+ *
+ *  Prerequisites: A pre-populated DynamoDB table.
+ *
+ *  Instructions for populating a table with sample data can be found at:
+ *  https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/SampleData.html

```

Comment:
*  For instructions on populating a table with sample data, see
 *  https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/SampleData.html.